x/ecocredit/server/core/seal_batch.go

```diff
@@ -3,9 +3,41 @@ package core
 import (
 	"context"
 
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
 	"github.com/regen-network/regen-ledger/x/ecocredit/core"
 )
 
+// SealBatch sets the Open field to false in a batch IFF the requester address matches the batch issuer address.
+// This method is essentially a no-op for batches which already have Open set to false.
 func (k Keeper) SealBatch(ctx context.Context, req *core.MsgSealBatch) (*core.MsgSealBatchResponse, error) {
-	panic("implement me")
+	issuer, err := sdk.AccAddressFromBech32(req.Issuer)
+	if err != nil {
+		return nil, err
+	}
+
+	batch, err := k.stateStore.BatchTable().GetByDenom(ctx, req.BatchDenom)
+	if err != nil {
+		return nil, sdkerrors.ErrInvalidRequest.Wrapf("could not get batch with denom %s: %s", req.BatchDenom, err.Error())
+	}
+
+	if !batch.Open {
+		return &core.MsgSealBatchResponse{}, nil
```

Comment:
Can we add a test for this?

Reply:
https://github.com/regen-network/regen-ledger/pull/1060/commits/578cb0e18e0313802d4aad48f09b54e502f0c21a